vod/hls/m3u8_builder.c

```diff
@@ -467,9 +468,15 @@ m3u8_builder_build_index_playlist(
 	segment_length = sizeof("#EXTINF:.000,\n") - 1 + vod_get_int_print_len(vod_div_ceil(duration_millis, 1000)) +
 		segments_base_url->len + conf->segment_file_name_prefix.len + 1 + vod_get_int_print_len(last_segment_index) + name_suffix.len;
 
+	int playlist_type_header_size = media_set->type == MEDIA_SET_VOD ?
+		sizeof(M3U8_HEADER_VOD) :
+		media_set->is_live_event ?
+		sizeof(M3U8_HEADER_EVENT) :
+		0;
+
 	result_size =
 		sizeof(M3U8_HEADER_PART1) + VOD_INT64_LEN +
-		sizeof(M3U8_HEADER_VOD) +
+		playlist_type_header_size +
```

Comment:
just use M3U8_HEADER_EVENT here, not trying to be exact (e.g. before the change, I always added vod, also in live)